### PR TITLE
G1 2024 v6 #14235

### PR DESCRIPTION
**Pull request by a22ricni**
Now it is possible to center the camera when there is no element on the screen. Can test it by have element on the screen and press the reset button to see if the camera is on the default position or not (it should only remove all the element). And if there is no element on screen then you should be able to center the camera to default position.

### CODE REVIEW

**Review by a22cazlu — CHANGES_REQUESTED** [review]
As i understand it the camera should focus at origo when there is no elements in the diagram and I press "Reset view". And when I test it that dosen't happened.

see video:

[screen-capture (4).webm](https://github.com/HGustavs/LenaSYS/assets/131054714/d08dc915-fba4-4c9e-ac9a-34f2ce1a7999)

**Review by a21liltr — CHANGES_REQUESTED** [review]
## Please read the issue again. Camera should reset when diagram is empty.

Your solution does not include a reset to the camera when pressing **reset view**.

https://github.com/HGustavs/LenaSYS/assets/129370135/aaaae487-ff27-4e1e-b8d9-a5f9e255b614

Also code should be in DuggaSys > diagram > camera.js

**Comment by a22ricni** [comment]
Should now work when "reset view" button is clicked if no element is on screen.

File: DuggaSys/diagram/camera.js
Comment (a21liltr) [inline DuggaSys/diagram/camera.js:1]: NOT OK: Duplicated code.

centerToOrigo is 90% duplicated code from centerCamera.

Remove the duplicated code in centerCamera and add centerToOrigo to centerCamera to use instead.
Make use of arguments to take in parameter to center to either origo or to the middle of the diagram with objects.
(Whether you use variables or bools to determine the recentering is up to you.)

**Review by a21liltr — CHANGES_REQUESTED** [review]
See comment on file below.

**Review by a22cazlu — CHANGES_REQUESTED** [review]
a21liltr mentions the issue in the code where there is duplicated code. But the diagram seems to work and the reset view now works when there is no elements in the diagram.

**Comment by a22ricni** [comment]
I have removed the duplicated code and hopefully made it more clean.

**Review by a22cazlu — APPROVED** [review]
What I can see it works. No meaning for me to request changes when a21liltr already done it.

**Review by a21liltr — APPROVED** [review]
Lgtm

**Pull request merged**